lib/beaker/hypervisor/openstack.rb

```diff
@@ -115,13 +115,28 @@ def provision
         # Associate a public IP to the server
         # Create if there are no floating ips available
         #
-        ip = @compute_client.addresses.find { |ip| ip.instance_id.nil? }
-        if ip.nil?
-          @logger.debug "Creating IP for #{host.name} (#{host[:vmhostname]})"
-          ip = @compute_client.addresses.create
+        # Do we already have an address?
+        @logger.debug vm.addresses
+
+        begin
+          if vm.addresses[@options[:openstack_network]]
+            address = vm.addresses[@options[:openstack_network]].map{ |network| network['addr'] }.first
+          end
+        rescue NoMethodError
+          @logger.debug "No current address retrievable from OpenStack data"
+        end
+        unless address
+          ip = @compute_client.addresses.find { |ip| ip.instance_id.nil? }
+
+          if ip.nil?
+            @logger.debug "Creating IP for #{host.name} (#{host[:vmhostname]})"
+            ip = @compute_client.addresses.create
+          end
+          ip.server = vm
+          address = ip.ip
         end
-        ip.server = vm
```

Comment:
ip.server used to be set for all cases - now it will not be set if the first being-rescue block is successful.  Is that the desired state here?